Client/core/CMainMenu.cpp

```diff
@@ -981,6 +1015,68 @@ bool CMainMenu::OnQuitButtonClick(CGUIElement* pElement)
     return true;
 }
 
+bool CMainMenu::OnSocialButtonHover(CGUIElement* pElement)
+{
+    pElement->SetAlpha(0.8f);
+
+    return true;
+}
+
+bool CMainMenu::OnSocialButtonUnhover(CGUIElement* pElement)
+{
+    pElement->SetAlpha(0.35f);
+
+    return true;
+}
+
+bool CMainMenu::OnWebsiteButtonClick(CGUIElement* pElement)
+{
+    // Visit the website url
+    ShellExecuteNonBlocking("open", "https://multitheftauto.com/");
+
+    return true;
+}
+
+bool CMainMenu::OnDiscordButtonClick(CGUIElement* pElement)
+{
+    // Visit the discord url
+    ShellExecuteNonBlocking("open", "https://multitheftauto.com/discord");
+
+    return true;
+}
+
+bool CMainMenu::OnTwitterButtonClick(CGUIElement* pElement)
+{
+    // Visit the twitter url
+    ShellExecuteNonBlocking("open", "https://twitter.com/MTAQA");
+
+    return true;
+}
+
+bool CMainMenu::OnFacebookButtonClick(CGUIElement* pElement)
+{
+    // Visit the facebook url
+    ShellExecuteNonBlocking("open", "https://www.facebook.com/multitheftauto");
+
+    return true;
+}
+
+bool CMainMenu::OnYoutubeButtonClick(CGUIElement* pElement)
+{
+    // Visit the youtube url
+    ShellExecuteNonBlocking("open", "https://www.youtube.com/user/MTAQA");
+
+    return true;
+}
+
+bool CMainMenu::OnGithubButtonClick(CGUIElement* pElement)
+{
+    // Visit the github url
+    ShellExecuteNonBlocking("open", "https://github.com/multitheftauto/mtasa-blue");
+
+    return true;
+}
```

Comment:
There is probably a nicer solution than having many functions. E.g. setting some data on the element that maps to some index, and then you can grab the link out of `socialButtonDatas`.

Reply:
Probably yes.
iirc `SubscribeEvent` takes a lambda as well (needs testing tho), if thats the case we dont even need these static functions